db/compaction_job.cc

```diff
@@ -747,6 +747,14 @@ void CompactionJob::ProcessKeyValueCompaction(SubcompactionState* sub_compact) {
       shutting_down_));
   auto c_iter = sub_compact->c_iter.get();
   c_iter->SeekToFirst();
+  if (c_iter->Valid()) {
+    // ShouldStopBefore() maintains state based on keys processed so far. The
+    // compaction loop always calls it on the "next" key, thus won't tell it the
+    // first key. So we do that here.
+    bool should_stop = sub_compact->ShouldStopBefore(
+      c_iter->key(), sub_compact->current_output_file_size);
+    assert(!should_stop);
```

Comment:
Oh, would you rather ignore the return value? Because I think there's no reasonable way for us to handle terminating files before keys are added, while there are still keys remaining in the compaction input.